types/bluebird/index.d.ts

```diff
@@ -58,7 +58,43 @@ declare class Bluebird<R> implements PromiseLike<R>, Bluebird.Inspection<R> {
    * The new promise will be rejected or resolved depending on the passed `fulfilledHandler`, `rejectedHandler` and the state of this promise.
    */
   // Based on PromiseLike.then, but returns a Bluebird instance.
-  then<U>(onFulfill?: (value: R) => Resolvable<U>, onReject?: (error: any) => Resolvable<U>): Bluebird<U>; // For simpler signature help.
+  then<T1, T2, T3, T4, T5>(onFulfill?: (value: R) => [
+      Resolvable<T1>,
+      Resolvable<T2>,
+      Resolvable<T3>,
+      Resolvable<T4>,
+      Resolvable<T5>
+  ], onReject?: (error: any) => [
+      Resolvable<T1>, 
+      Resolvable<T2>,
+      Resolvable<T3>,
+      Resolvable<T4>,
+      Resolvable<T5>
+  ]): Bluebird<[T1, T2, T3, T4, T5]>;
+  then<T1, T2, T3, T4>(onFulfill?: (value: R) => [
+      Resolvable<T1>,
+      Resolvable<T2>,
+      Resolvable<T3>,
+      Resolvable<T4>
+  ], onReject?: (error: any) => [
+      Resolvable<T1>,
+      Resolvable<T2>,
+      Resolvable<T3>,
+      Resolvable<T4>
+  ]): Bluebird<[T1, T2, T3, T4]>;
+  then<T1, T2, T3>(onFulfill?: (value: R) => [
+      Resolvable<T1>,
+      Resolvable<T2>,
+      Resolvable<T3>
+  ], onReject?: (error: any) => [Resolvable<T1>, Resolvable<T2>, Resolvable<T3>]): Bluebird<[T1, T2, T3]>;
```

Comment:
Nit: You forgot to add line breaks here.
Since it's only a very minor stylistic issue I'll approve your PR regardless. Of course it'd be nice if you fixed it though. 😄